### ISSUE TITLE
[question] layout and --test-build-folder

### ISSUE THREAD

**Issue by Hopobcn** [conan-io/conan#10201]
Hi, 

TLTR:
If a `test_package` conanfile uses `layout()` is `--test-bulid-folder` ignored?

-----

I'm trying to write a `test_package` parametrized by generator (to test `CMakeDeps` & `cmake_find_package`).

```
test_package/
├── CMakeDeps
│   └── conanfile.py
├── cmake_find_package
│   └── conanfile.py
├── CMakeLists.txt
└── test_package.cpp
```

`test_package/cmake_find_package/conanfile.py` is a pretty standard test package with the particularity that `source_folder` is declared as one level up to reuse the same `CMakeLists.txt`/`test_package.cpp` between both tests:
```
from conans import ConanFile, CMake, tools
import os

class LoggingTestConan(ConanFile):
    settings = "os", "compiler", "build_type", "arch"
    generators = "cmake_find_package"

    def build(self):
        cmake = CMake(self)
        cmake.configure(source_folder="..")
        cmake.build()

    def test(self):
        if not tools.cross_building(self.settings):
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path, run_environment=True)
```
`conan test test_package/cmake_find_package foo/1.2.3@ -tbf build_old` generates a `build_old` folder.

In the `CMakeDeps` case:
```
from conan import ConanFile
from conan.tools.cmake import CMake
from conan.tools.layout import cmake_layout
from conans import tools
import os

class LoggingTestConan(ConanFile):
    settings = "os", "compiler", "build_type", "arch"
    generators = "CMakeToolchain", "CMakeDeps"

    def layout(self):
        cmake_layout(self)
        self.folders.source = ".."

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def test(self):
        if not tools.cross_building(self.settings):
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path, run_environment=True)

```
`conan test test_package/CMakeDeps foo/1.2.3@ -tbf build_new` does not generate any `build_new` folder. Instead, it follows what the `cmake_layout` imposes:
```
test_package/CMakeDeps/
├── cmake-build-release
│   └── ...
├── conanbuildinfo.txt
├── conanfile.py
├── conaninfo.txt
├── conan.lock
└── graph_info.json
```
I this expected behavior? If it is, could `conan test` detect if you're using `layout` and warn you about this?

Also, is there any way to reduce file cluttering (`conanbuildinfo.txt`, `conaninfo.txt`, `conan.lock`, `graph_info.json`)  in  `test_package` folder? (they're being generated outside the build folder)
With `conan install` we can control this via `--install-folder` but I didn't find a way with `conan test`. 

Thanks,

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @Hopobcn 

At the moment the ``layout()`` feature always references to the conanfile.py position. We are in the middle of a change to allow following local folders different to the conanfile.py one (that would included the test one too).

This is the first PR towards this goal: https://github.com/conan-io/conan/pull/10207, and https://github.com/conan-io/conan/pull/10180 would be important too.

**Comment by ericriff** [comment]
I just came across this myself. Is this still the case? Even if I pass `--test-build-folder=/tmp/asd` to a recipe that uses `layout()` the package gets built on `library/all/test_package/build`.

**Comment by ericriff** [comment]
I'm still trying to figure out a workaround for this BTW https://github.com/conan-io/conan/issues/12293

**Comment by ericriff** [comment]
Hi @memsharded 
I see that both PRs you linked above have been merged, yet I still can't use `--test-bulid-folder` when the test package uses `layout()`. It just gets ignored. 

Am I missing something? The issue I linked above is very annoying for us and I'm trying to find a workaround for it. We compile the same recipes for a bunch of archs at the same time, in parallel, and the temp files of the test package end up conflicting. This wasn't a problem before since the folder structure used to be `test_pacakge/build/hash`, but now that the `hash` is gone all the builds try to put the cmake files on the same place.

**Comment by memsharded** [comment]
Hi @ericriff 

Sorry for the delay. The ``--test-build-folder`` has still not be implemented in 2.0, and it is not clear if it will be. I will try to discuss the case with the team and see if there is something we can do for beta.8.

In the meantime, I am testing in https://github.com/conan-io/conan/pull/12864 something that could help here. 
The ``tools.cmake.cmake_layout:build_folder_vars`` conf allows to customize the layouts, including the ``test_package`` layout. So you only need to define this conf in the command line with ``-c`` and the folders will no longer conflict, each different configuration will have a different folder, named after the configuration. That should avoid concurrency errors because of building them in parallel. If you could please try it, let us know how it goes.

**Comment by memsharded** [comment]
The thing to discuss for beta.8 would be:

- Do we need the equivalent of ``--output-folder``, that applies for ``test_package``?
- Can we reuse the ``--output-folder`` meaning for ``test_package`` (as ``conan create`` will no longer need the local build folder)

**Comment by ericriff** [comment]
Is there a reason why we build the test packages inside the repo anyways? When my local cache gets big because there are old packages laying around, I can simply `rm -rf ~/.conan/data`. That kind of cleaning is not straightforward with the test packages build.

**Comment by ericriff** [comment]
Ok using `tools.cmake.cmake_layout:build_folder_vars` is a pretty good workaround. I believe I can cover most of our cases by doing 
```
-c tools.cmake.cmake_layout:build_folder_vars="['settings.arch', 'settings.compiler','settings.compiler.version', 'settings.shared']"
```
It is not perfect though. If I want to build the same packages but with different options in parallel, then I can't. You could say, then add the option to the list above. But the problem is that options are recipe-specific and I'd like to have a single, generic statement that I can include in my build scripts. 

I have to say I don't understand why the behavior changed from 1.x. Each package id having its own test package build folder made a lof of sense.

**Comment by memsharded** [comment]
> I have to say I don't understand why the behavior changed from 1.x. Each package id having its own test package build folder made a lof of sense.

The case of concurrent ``conan create`` was never really considered, because the Conan cache is explicitly not concurrent, so it is very possible that such concurrent create fails due to race conditions.

With that idea, Conan 2.0 is creating an intermediate ``test_output`` folder that is very easy to ``gitignore`` and that is removed at every ``conan create`` ``test_package`` step, to have a clean build. It is also possible to parameterize this in ``self.folders.test_output`` in the ``layout()`` method of ``test_package/conanfile.py`` to define custom output folders. 

One of the main reasons this was removed is that the previous approach was creating multiple arbitrary folders that were meaningless, it was not possible to traceback the folder name to the specific configuration that it was testing. And mostly it seemed that users wanted just a deterministic folder where to find some files in case something went wrong in the ``test_package`` step.

**Comment by memsharded** [comment]
We are ok in adding back some ``--test-output-folder`` argument, but I am thinking what other alternatives are possible

**Comment by ericriff** [comment]
The thing is, this fails even without parallel builds. See this example, I'm first running the test package for an aarch64 profile, then for x86. Since it uses the same build folderm there are linking errrors
```
$ conan test recipes/opengv/all/test_package opengv/cci.20200806@asd/asd -pr:b x86_64_gcc8 -pr:h aarch64 --build=missing
Configuration (profile_host):
[settings]
arch=aarch64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.sdkversion=SDK/aarch64-2.0@eriff/stable
compiler.version=11
os=Linux
os_build=Linux
[options]
[build_requires]
*: SDK/aarch64-2.0@eriff/stable
[env]
CHOST=aarch64-oe-linux
CONAN_CMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY
CONAN_CMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY
CONAN_CMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ONLY
CONAN_CMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER
CPATH=[/home/eriff/conan/.conanprofiles/../irstd/include]
[conf]
tools.gnu:host_triplet=aarch64-oe-linux

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
os_build=Linux
[options]
[build_requires]
[env]
CPATH=[/home/eriff/conan/.conanprofiles/../irstd/include]
[conf]
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True

opengv/cci.20200806@asd/asd (test package): Installing package
Requirements
    eigen/3.3.8@eriff/stable from local cache - Cache
    opengv/cci.20200806@asd/asd from local cache - Cache
Packages
    eigen/3.3.8@eriff/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    opengv/cci.20200806@asd/asd:d253ff5fd0a464e54bf1f77a47f89b82b7c91a8e - Cache
Build requirements
    SDK/aarch64-2.0@eriff/stable from local cache - Cache
    Toolchain/aarch64-11.3.1.irbt1@eriff/stable from local cache - Cache
Build requirements packages
    SDK/aarch64-2.0@eriff/stable:e55ffc5d2aaae31474cd4ef086293217c79b14a2 - Cache
    Toolchain/aarch64-11.3.1.irbt1@eriff/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache

Cross-build from 'Linux:x86_64' to 'Linux:aarch64'
Installing (downloading, building) binaries...
Toolchain/aarch64-11.3.1.irbt1@eriff/stable: Already installed!
eigen/3.3.8@eriff/stable: Already installed!
SDK/aarch64-2.0@eriff/stable: Already installed!
opengv/cci.20200806@asd/asd: Already installed!
opengv/cci.20200806@asd/asd (test package): Applying build-requirement: SDK/aarch64-2.0@eriff/stable
opengv/cci.20200806@asd/asd (test package): Applying build-requirement: Toolchain/aarch64-11.3.1.irbt1@eriff/stable
opengv/cci.20200806@asd/asd (test package): Generator 'CMakeDeps' calling 'generate()'
opengv/cci.20200806@asd/asd (test package): Generator txt created conanbuildinfo.txt
opengv/cci.20200806@asd/asd (test package): Generator 'CMakeToolchain' calling 'generate()'
opengv/cci.20200806@asd/asd (test package): Preset 'release' added to CMakePresets.json. Invoke it manually using 'cmake --preset release'
opengv/cci.20200806@asd/asd (test package): If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE=/home/eriff/conan/recipes/opengv/all/test_package/build/generators/conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release'
opengv/cci.20200806@asd/asd (test package): Generator 'VirtualRunEnv' calling 'generate()'
opengv/cci.20200806@asd/asd (test package): Aggregating env generators
opengv/cci.20200806@asd/asd (test package): Generated conaninfo.txt
opengv/cci.20200806@asd/asd (test package): Generated graphinfo
Using lockfile: '/home/eriff/conan/recipes/opengv/all/test_package/build/generators/conan.lock'
Using cached profile from lockfile
opengv/cci.20200806@asd/asd (test package): Calling build()
opengv/cci.20200806@asd/asd (test package): CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/eriff/conan/recipes/opengv/all/test_package/build/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/eriff/conan/recipes/opengv/all/test_package/."
-- Using Conan toolchain: /home/eriff/conan/recipes/opengv/all/test_package/build/generators/conan_toolchain.cmake
-- The CXX compiler identification is GNU 11.3.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/eriff/conan/.conan/data/Toolchain/aarch64-11.3.1.irbt1/eriff/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/gcc-linaro-11.3.1-11.3.1.irbt1-x86_64_aarch64-none-linux-gnu/bin/aarch64-none-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'opengv'
-- Conan: Component target declared 'Eigen3::Eigen'
-- Configuring done
-- Generating done
-- Build files have been written to: /home/eriff/conan/recipes/opengv/all/test_package/build/Release
opengv/cci.20200806@asd/asd (test package): CMake command: cmake --build "/home/eriff/conan/recipes/opengv/all/test_package/build/Release" '--' '-j12'
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
[100%] Linking CXX executable test_package
[100%] Built target test_package
opengv/cci.20200806@asd/asd (test package): Running test()





$ conan test recipes/opengv/all/test_package opengv/cci.20200806@asd/asd -pr:b x86_64_gcc8 -pr:h x86_64_gcc8 --build=missing
Configuration (profile_host):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
os_build=Linux
[options]
[build_requires]
[env]
[conf]
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
os_build=Linux
[options]
[build_requires]
[env]
[conf]
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True

opengv/cci.20200806@asd/asd (test package): Installing package
Requirements
    eigen/3.3.8@eriff/stable from local cache - Cache
    opengv/cci.20200806@asd/asd from local cache - Cache
Packages
    eigen/3.3.8@eriff/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    opengv/cci.20200806@asd/asd:ce5a6f5af3cdc32827c191d83e1eb6be492f49c4 - Cache

Installing (downloading, building) binaries...
eigen/3.3.8@eriff/stable: Already installed!
opengv/cci.20200806@asd/asd: Already installed!
opengv/cci.20200806@asd/asd (test package): Generator txt created conanbuildinfo.txt
opengv/cci.20200806@asd/asd (test package): Generator 'CMakeToolchain' calling 'generate()'
opengv/cci.20200806@asd/asd (test package): Preset 'release' added to CMakePresets.json. Invoke it manually using 'cmake --preset release'
opengv/cci.20200806@asd/asd (test package): If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE=/home/eriff/conan/recipes/opengv/all/test_package/build/generators/conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release'
opengv/cci.20200806@asd/asd (test package): Generator 'VirtualRunEnv' calling 'generate()'
opengv/cci.20200806@asd/asd (test package): Generator 'CMakeDeps' calling 'generate()'
opengv/cci.20200806@asd/asd (test package): Aggregating env generators
opengv/cci.20200806@asd/asd (test package): Generated conaninfo.txt
opengv/cci.20200806@asd/asd (test package): Generated graphinfo
Using lockfile: '/home/eriff/conan/recipes/opengv/all/test_package/build/generators/conan.lock'
Using cached profile from lockfile
opengv/cci.20200806@asd/asd (test package): Calling build()
opengv/cci.20200806@asd/asd (test package): CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/eriff/conan/recipes/opengv/all/test_package/build/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/eriff/conan/recipes/opengv/all/test_package/."
-- Using Conan toolchain: /home/eriff/conan/recipes/opengv/all/test_package/build/generators/conan_toolchain.cmake
-- Conan: Target declared 'opengv'
-- Conan: Component target declared 'Eigen3::Eigen'
-- Configuring done
-- Generating done
-- Build files have been written to: /home/eriff/conan/recipes/opengv/all/test_package/build/Release
opengv/cci.20200806@asd/asd (test package): CMake command: cmake --build "/home/eriff/conan/recipes/opengv/all/test_package/build/Release" '--' '-j12'
Consolidate compiler generated dependencies of target test_package
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
[100%] Linking CXX executable test_package
/home/eriff/conan/.conan/data/Toolchain/aarch64-11.3.1.irbt1/eriff/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/gcc-linaro-11.3.1-11.3.1.irbt1-x86_64_aarch64-none-linux-gnu/bin/../lib/gcc/aarch64-none-linux-gnu/11.3.1/../../../../aarch64-none-linux-gnu/bin/ld: /home/eriff/conan/.conan/data/opengv/cci.20200806/asd/asd/package/ce5a6f5af3cdc32827c191d83e1eb6be492f49c4/lib/libopengv.a(roots.cpp.o): Relocations in generic ELF (EM: 62)
/home/eriff/conan/.conan/data/Toolchain/aarch64-11.3.1.irbt1/eriff/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/gcc-linaro-11.3.1-11.3.1.irbt1-x86_64_aarch64-none-linux-gnu/bin/../lib/gcc/aarch64-none-linux-gnu/11.3.1/../../../../aarch64-none-linux-gnu/bin/ld: /home/eriff/conan/.conan/data/opengv/cci.20200806/asd/asd/package/ce5a6f5af3cdc32827c191d83e1eb6be492f49c4/lib/libopengv.a(roots.cpp.o): Relocations in generic ELF (EM: 62)
/home/eriff/conan/.conan/data/Toolchain/aarch64-11.3.1.irbt1/eriff/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/gcc-linaro-11.3.1-11.3.1.irbt1-x86_64_aarch64-none-linux-gnu/bin/../lib/gcc/aarch64-none-linux-gnu/11.3.1/../../../../aarch64-none-linux-gnu/bin/ld: /home/eriff/conan/.conan/data/opengv/cci.20200806/asd/asd/package/ce5a6f5af3cdc32827c191d83e1eb6be492f49c4/lib/libopengv.a(roots.cpp.o): Relocations in generic ELF (EM: 62)
/home/eriff/conan/.conan/data/Toolchain/aarch64-11.3.1.irbt1/eriff/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/gcc-linaro-11.3.1-11.3.1.irbt1-x86_64_aarch64-none-linux-gnu/bin/../lib/gcc/aarch64-none-linux-gnu/11.3.1/../../../../aarch64-none-linux-gnu/bin/ld: /home/eriff/conan/.conan/data/opengv/cci.20200806/asd/asd/package/ce5a6f5af3cdc32827c191d83e1eb6be492f49c4/lib/libopengv.a(roots.cpp.o): Relocations in generic ELF (EM: 62)
/home/eriff/conan/.conan/data/Toolchain/aarch64-11.3.1.irbt1/eriff/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/gcc-linaro-11.3.1-11.3.1.irbt1-x86_64_aarch64-none-linux-gnu/bin/../lib/gcc/aarch64-none-linux-gnu/11.3.1/../../../../aarch64-none-linux-gnu/bin/ld: /home/eriff/conan/.conan/data/opengv/cci.20200806/asd/asd/package/ce5a6f5af3cdc32827c191d83e1eb6be492f49c4/lib/libopengv.a(roots.cpp.o): Relocations in generic ELF (EM: 62)
/home/eriff/conan/.conan/data/Toolchain/aarch64-11.3.1.irbt1/eriff/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/gcc-linaro-11.3.1-11.3.1.irbt1-x86_64_aarch64-none-linux-gnu/bin/../lib/gcc/aarch64-none-linux-gnu/11.3.1/../../../../aarch64-none-linux-gnu/bin/ld: /home/eriff/conan/.conan/data/opengv/cci.20200806/asd/asd/package/ce5a6f5af3cdc32827c191d83e1eb6be492f49c4/lib/libopengv.a(roots.cpp.o): Relocations in generic ELF (EM: 62)
/home/eriff/conan/.conan/data/Toolchain/aarch64-11.3.1.irbt1/eriff/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/gcc-linaro-11.3.1-11.3.1.irbt1-x86_64_aarch64-none-linux-gnu/bin/../lib/gcc/aarch64-none-linux-gnu/11.3.1/../../../../aarch64-none-linux-gnu/bin/ld: /home/eriff/conan/.conan/data/opengv/cci.20200806/asd/asd/package/ce5a6f5af3cdc32827c191d83e1eb6be492f49c4/lib/libopengv.a: error adding symbols: file in wrong format
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/test_package.dir/build.make:98: test_package] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/test_package.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
ERROR: opengv/cci.20200806@asd/asd (test package): Error in build() method, line 21
	cmake.build()
	ConanException: Error 2 while executing cmake --build "/home/eriff/conan/recipes/opengv/all/test_package/build/Release" '--' '-j12'
```

**Comment by memsharded** [comment]
Yes, yes, I know it can fail in 1.X when changing configurations and it is painful. I meant this will not error in 2.0

**Comment by memsharded** [comment]
We have discussed this, and we are trying to propose a solution in 1.58 scope based on the following ideas:
- A simple way to define a ``layout()`` for test_package, telling it is a ``test_package`` layout
- This new layout will generate different folders for every different configuration, including also a timestamp, just in case, to completely avoid collisions. 
- Conan will never remove these folders automatically (as 2.0 is doing with ``test_output``)
- There will be an easy way to gitignore these folders
- There will be a conf in ``global.conf`` to allow doing these builds in other external tmp folders to avoid polluting the current ``test_package``, and not needing to define it as a cli arg.

**Comment by memsharded** [comment]
We have been trying different things towards this goal, and failed miserably 😞 :

- https://github.com/conan-io/conan/pull/12991 tries to let the original sha-based folder as unique identifier, even when ``layout()`` method exists. This completely breaks layouts like Visual, Bazel or Autotools (in some cases), that cannot relocate their build folder as it is hardcoded in the build system.
- https://github.com/conan-io/conan/pull/12981 only achieved some progress by a new opt-in ``conf`` to define the folders for tests, but in a different order, so it is not easily to have something that upgrades to 2.0 cleanly, and do not solve the easy gitignoration of folders or the global relocation of the test_package output folder in build systems that support it.

The alternatives for concurrent and better usage could be:
- Keep the ``layout()`` and use ``git clean -xdf`` or similar cleaning strategy to remove the previous test_package/build folders. This is probably good for most developers.
- To drop the ``test_package/conanfile.py`` ``layout()``, if the issue is a big pain to do concurrent test in CI, this will use the complete unique hash per configuration. If we want to make it compatible and ready for 2.0, this is a problem. The best approach is to use different test_package and test_package_v2 folders, and launch the second with conan create ... -tf=test_package_v2 in Conan 2.0. This is the approach we are using in ConanCenter.

We have started to at least remove these pain points for 2.0 in https://github.com/conan-io/conan/pull/12992 (still some PRs to follow), but at the moment seems impossible to have something for 1.X.

**Comment by memsharded** [comment]
Conan 2.0 was released now 1 year ago, as it wasn't possible to come up with a reasonable solution in 1.X, I am closing this ticket as solved in Conan 2.0. Do not hesitate to open new tickets for any further issue, thanks for the feedback!
